### DISCUSSION TITLE
Usefulness of matrices module

### DISCUSSION THREAD

**Discussion by andrii0lomakin** [beehive-lab/TornadoVM#408] [Q&A]
Hi.
I am not sure that I understand the usefulness of the matrices module.

Let us look at https://github.com/beehive-lab/TornadoVM/blob/master/tornado-matrices/src/main/java/uk/ac/manchester/tornado/matrix/MatrixMath.java#L47.
In this implementation, there are neither `@Parallel` annotations nor Kernel API. 

What is the purpose of this class? Could you comment on it?

**Comment by jjfumero** [comment]
Hi, this class is used by another package for Computer Vision:

https://github.com/beehive-lab/kfusion-tornadovm

The goal is to be invoked by this of applications in which kernels are already parallelized and some of the functions in this package might not be parallelized. However, as you can see here, `sgemm` is parallel:
https://github.com/beehive-lab/TornadoVM/blob/master/tornado-matrices/src/main/java/uk/ac/manchester/tornado/matrix/MatrixMath.java#L80-L102

Do you have something in mind? What are you trying to build?

**Comment by andrii0lomakin** [comment]
Currently, I am working on the POC of transformer implementation, so the matrices module caught my eye. I will create my own implementations of matrix operations then. Thank you for your responses. They are always very quick.

**Reply by jjfumero** [reply]
That's really interesting. Keep us posted.

**Reply by andrii0lomakin** [reply]
With pleasure :-)

**Comment by kotselidis** [comment]
Hi, We also have a slack channel if you would like to join: https://join.slack.com/t/tornadovmcommunity/shared_invite/zt-1sx0elmm8-hc~CUPO5xEW0vTaZKIXASg
